src/models/tag.ts

```diff
@@ -243,6 +249,93 @@ export class TagModel {
   //   public getSuggestedTags() {}
   //   public getBySave() {}
   //   public removeTagSaveConnections {}
+
+  /**
+   * Bulk update method for creating and deleting tags associated to a Save.
+   * All SaveIds passed to this function must be valid; if any are not
+   * found in the user's saves, the entire operation will be rolled
+   * back and NotFound payload returned.
+   * If attempting to delete a tag that does not exist,
+   * the method is a "no-op" and will not result in an error.
+   * @param updates payload of deletes and creates for a given saveId
+   * @param timestamp timestamp for when the bulk operation occurred
+   * @returns the saves that were updated, and any errors that occurred
+   */
+  public async batchUpdateTagConnections(
+    input: SaveUpdateTagsInputGraphql[],
+    timestamp: Date,
+    path: GraphQLResolveInfo['path']
+  ): Promise<SaveWriteMutationPayload> {
+    const batchUpdate = this.formatBatchUpdateInput(input);
+    const { updated, missing } = await this.tagService.batchUpdateTags(
+      batchUpdate,
+      timestamp
+    );
+    let save: PocketSave[] = [];
+    if (updated.length) {
+      const ids = updated.map((update) => update.toString());
+      save = await this.context.models.pocketSave.getManyById(ids);
```

Comment:
ah why the above that I thought was https://getpocket.atlassian.net/browse/INFRA-810 was setup - i get it